xbmc/addons/AddonRepos.cpp

```diff
@@ -270,3 +271,59 @@ bool CAddonRepos::GetLatestAddonVersion(const std::string& addonId, AddonPtr& re
 
   return true;
 }
+
+bool CAddonRepos::FindDependency(const std::string& dependsId,
+                                 const AddonPtr& parent,
+                                 AddonPtr& dependencyToInstall,
+                                 std::shared_ptr<CRepository>& repoForDep) const
+{
+  const AddonRepoUpdateMode updateMode =
+      CAddonSystemSettings::GetInstance().GetAddonRepoUpdateMode();
+
+  bool hasOfficialVersion =
+      GetLatestVersionByMap(dependsId, m_latestOfficialVersions, dependencyToInstall);
```

Comment:
This will be true for any dependency from an official repo so the next condition will never be entered if the dep is official.